File: system-test/dotnet/Dsl.Core/UseCase/UseCases/GoToMyShop.cs
using Driver.Port;
using Dsl.Core.UseCase.UseCases.Base;
using Driver.Port.Dtos.Error;
using Common;
using Dsl.Core.Shared;

namespace Dsl.Core.UseCase.UseCases;

public class GoToMyShop : BaseMyShopUseCase<VoidValue, VoidVerification>
{
    public GoToMyShop(IMyShopDriver driver, UseCaseContext context)
        : base(driver, context)
    {
    }

    public override async Task<MyShopUseCaseResult<VoidValue, VoidVerification>> Execute()
    {
        var result = await _driver.GoToMyShopAsync();

        return new MyShopUseCaseResult<VoidValue, VoidVerification>(
            result,
            _context,
            (response, ctx) => new VoidVerification(response, ctx));
    }
}

File: system/multitier/backend-java/src/testSupport/java/com/mycompany/myshop/backend/support/core/usecase/usecases/GoToMyShop.java
package com.mycompany.myshop.backend.support.core.usecase.usecases;

import static org.assertj.core.api.Assertions.assertThat;

import com.mycompany.myshop.backend.support.BackendDriver;
import com.mycompany.myshop.backend.support.core.shared.UseCaseResult;
import com.mycompany.myshop.backend.support.core.shared.VoidVerification;
import com.mycompany.myshop.backend.support.core.usecase.usecases.base.BaseMyShopUseCase;
import org.springframework.http.HttpStatus;

/**
 * The liveness probe behind {@code assume().myShop().shouldBeRunning()}: {@code GET /health} must
 * answer {@code 200 OK} with {@code status: UP}. Both halves of that contract are asserted by
 * {@code shouldSucceed()} — the status by {@link UseCaseResult}, the body here.
 */
public class GoToMyShop extends BaseMyShopUseCase<Void, VoidVerification> {

    public GoToMyShop(BackendDriver driver) {
        super(driver);
    }

    @Override
    public UseCaseResult<Void, VoidVerification> execute() {
        var response = driver.checkHealth();

        return new UseCaseResult<>(
            response.getStatusCode(),
            HttpStatus.OK,
            null,
            () -> {
                assertThat(response.getBody()).as("health body").containsEntry("status", "UP");
                return null;
            },
            null,
            VoidVerification::new);
    }
}
